test/functional/hooks.js

```diff
@@ -41,38 +43,120 @@ describe('library hooks', () => {
           'dynamicName', dynamicValue);
     }
 
+    function stubOut(mutator) {
+      stub = sandbox.stub();
+      mutators.attributes[mutator] = stub;
+    }
+
     beforeEach(() => {
-      sandbox.spy(attributes, 'applyAttr');
+      allSpy = sandbox.spy(mutators.attributes, '__all');
     });
-  
-    it('should be called for static attributes', () => {
-      patch(container, render, 'dynamicValue');
-      var el = container.childNodes[0];
 
-      expect(attributes.applyAttr).calledWith(el, 'staticName', 'staticValue');
+    afterEach(() => {
+      for (var mutator in mutators.attributes) {
+        if (mutator !== '__all') {
+          mutators.attributes[mutator] = null;
+        }
+      }
     });
 
-    it('should be called for dynamic attributes', () => {
-      patch(container, render, 'dynamicValue');
-      var el = container.childNodes[0];
 
-      expect(attributes.applyAttr).calledWith(el, 'dynamicName', 'dynamicValue');
+    describe('for static attributes', () => {
+      it('should call specific setter', () => {
+        stubOut('staticName');
+
+        patch(container, render, 'dynamicValue');
+        var el = container.childNodes[0];
+
+        expect(stub).to.have.been.calledOnce;
+        expect(stub).to.have.been.calledWith(el, 'staticName', 'staticValue');
+      });
+
+      it('should call generic setter', () => {
+        patch(container, render, 'dynamicValue');
+        var el = container.childNodes[0];
+
+        expect(allSpy).to.have.been.calledWith(el, 'staticName', 'staticValue');
+      });
+
+      it('should prioritize specific setter over generic', () => {
+        stubOut('staticName');
+
+        patch(container, render, 'dynamicValue');
+        var el = container.childNodes[0];
+
+        expect(stub).to.have.been.calledOnce;
+        expect(allSpy).to.have.been.calledOnce;
+        expect(allSpy).to.have.been.calledWith(el, 'dynamicName', 'dynamicValue');
```

Comment:
Nope, I'm testing that `allSpy` has not been called for the static attribute. But, it'll get called for the dynamic one, so I'm asserting the single call is for the dynamic attribute.